### DISCUSSION TITLE
Required assistance

### DISCUSSION THREAD

**Discussion by Vaishnavi-Ind** [coders-school/cpp-fundamentals#1400] [General] [answered]
How can we ensure a dynamically allocated memory in a C++ program is properly deallocated to avoid memory leaks?

**Comment by Shriharsh-Deshmukh** [comment, marked as answer]
To avoid memory leaks when using dynamically allocated memory in C++:

To avoid memory leaks when using dynamically allocated memory in C++:

**1. Always pair new with delete and new[] with delete[].** For example:
  
   int* num = new int(5); // Allocate memory
   delete num;            // Free memory

   int* arr = new int[10]; // Allocate an array
   delete[] arr;           // Free the array
  

**2. Use smart pointers like std::unique_ptr or std::shared_ptr because they automatically free the memory when no longer needed:**
   
   #include <memory>
   std::unique_ptr<int> num(new int(5)); // Automatically managed
  

This way, you don’t forget to release memory, and your program stays efficient and safe.

**Reply by Vaishnavi-Ind** [reply]
Ok and is it necessary to manually deallocate memory when using new in C++ to avoid memory leaks?

**Reply by Shriharsh-Deshmukh** [reply]
> Ok and is it necessary to manually deallocate memory when using new in C++ to avoid memory leaks?

Yes!

**Reply by Vaishnavi-Ind** [reply]
Ok Thanks 👍
